packages/scripts/src/flux-emulator/ReferenceContractConfig.ts

```diff
@@ -0,0 +1,300 @@
+import axios from 'axios-observable'
+import { transformAndValidateSync } from 'class-transformer-validator'
+import { retryBackoff } from 'backoff-rxjs'
+import { Observable, of } from 'rxjs'
+import { catchError, map } from 'rxjs/operators'
+import console from 'console'
+import { AxiosRequestConfig } from 'axios'
+
+export class ReferenceContractConfig {
+  name: string
+  contractVersion: number
+  address: string
+  data: Record<string, unknown>
+  nodes: FeedNode[]
+  precision: number
+  deviationThreshold: number
+  symbol: string
+  path: string
+  status: string
+
+  constructor() {
+    this.name = ''
+    this.contractVersion = 0
+    this.address = ''
+    this.deviationThreshold = 0
+    this.data = {}
+    this.nodes = []
+    this.precision = 8
+    this.symbol = ''
+    this.path = ''
+    this.status = ''
+  }
+}
+
+export class FeedNode {
+  name: string
+  address: string
+  dataProviders: string[]
+  constructor() {
+    this.name = ''
+    this.address = ''
+    this.dataProviders = []
+  }
+}
+
+export interface K6Payload {
+  name: string
+  id: string
+  method: string
+  data: string
+}
+
+export type ReferenceContractConfigResponse = {
+  configs: ReferenceContractConfig[] | undefined
+}
+
+type ApiResponse = Record<string, unknown>[]
+const MAX_REQUESTS = 5
+const REQUEST_TIMEOUT_MS = 5000
+const RETRY_BACKOFF = {
+  resetOnSuccess: true,
+  initialInterval: 250,
+  maxRetries: MAX_REQUESTS - 1,
+  maxInterval: 2000,
+}
+
+/**
+ * Will fetch a config from the web
+ * @param {string} configUrl The url to a config
+ * @returns {ReferenceContractConfigResponse} The response data if it existed
+ */
+export const fetchConfigFromUrl = (
+  configUrl: string,
+): Observable<ReferenceContractConfigResponse> => {
+  let requestAttempt = 0
+
+  return axios.get<ApiResponse>(configUrl, { timeout: REQUEST_TIMEOUT_MS }).pipe(
+    map((res: { data: Record<string, unknown>[] }) => {
+      // console.log(res.data)
```

Comment:
Intentionally commented out? If so we could remove?

Reply:
Just missed during cleanup